packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp

```diff
@@ -300,9 +316,12 @@ namespace MueLu {
       // Compute matrix on bottom of generalized Rayleigh quotient
       DenseMatrix bottomMatrix(A_aggPart);
       MT matrixNorm = A_aggPart.normInf();
+
+      // Forward mode: Include a small perturbation to the bottom matrix to make it nonsingular
+      MT boost = (algo == ALG_FORWARD) ? (-1e-12*matrixNorm) : MT_ZERO;
```

Comment:
const ?

Reply:
Instead of using 1e-12, how about `1e4*Teuchos::ScalarTraits<MT>::eps()`?

Reply:
@lucbv Sure.

@Because that's what the original code had.  Will fix it though..,.